projects/opendr_ws/src/perception/scripts/object_tracking_3d_ab3dmot.py

```diff
@@ -76,44 +80,82 @@ def callback(self, data):
         point_cloud = self.bridge.from_ros_point_cloud(data)
         detection_boxes = self.detector.infer(point_cloud)
         tracking_boxes = self.learner.infer(detection_boxes)
-        ids = [tracking_box.id for tracking_box in tracking_boxes]
 
-        # Convert detected boxes to ROS type and publish
-        ros_boxes = self.bridge.to_ros_boxes_3d(detection_boxes, classes=["Car", "Van", "Truck", "Pedestrian", "Cyclist"])
         if self.detection_publisher is not None:
+            # Convert detected boxes to ROS type and publish
+            ros_boxes = self.bridge.to_ros_boxes_3d(detection_boxes, classes=["Car", "Van", "Truck", "Pedestrian", "Cyclist"])
             self.detection_publisher.publish(ros_boxes)
             rospy.loginfo("Published detection boxes")
 
-        ros_ids = Int32MultiArray()
-        ros_ids.data = ids
-
         if self.tracking_id_publisher is not None:
+            ids = [tracking_box.id for tracking_box in tracking_boxes]
+            ros_ids = Int32MultiArray()
+            ros_ids.data = ids
             self.tracking_id_publisher.publish(ros_ids)
             rospy.loginfo("Published tracking ids")
 
-if __name__ == "__main__":
-    # Automatically run on GPU/CPU
-    device = "cuda:0" if torch.cuda.is_available() else "cpu"
+    def listen(self):
+        """
+        Start the node and begin processing input data.
+        """
+        rospy.init_node('ab3dmot_tracking_3d', anonymous=True)
+        rospy.Subscriber(self.input_point_cloud_topic, ROS_PointCloud, self.callback, queue_size=1, buff_size=10000000)
+
+        rospy.loginfo("Object Tracking 3D Ab3dmot Node started.")
+        rospy.spin()
 
-    # initialize ROS node
-    rospy.init_node("opendr_voxel_detection_3d", anonymous=True)
-    rospy.loginfo("AB3DMOT node started")
 
-    input_point_cloud_topic = rospy.get_param(
-        "~input_point_cloud_topic", "/opendr/dataset_point_cloud"
-    )
-    temp_dir = rospy.get_param("~temp_dir", "temp")
-    detector_model_name = rospy.get_param("~detector_model_name", "tanet_car_xyres_16")
-    detector_model_config_path = rospy.get_param(
-        "~detector_model_config_path", os.path.join(
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-dn", "--detector_model_name", help="Name of the trained model",
+                        type=str, default="tanet_car_xyres_16")
```

Comment:
Added choices argument, it seems redundant but it shows up when calling help and avoids users providing wrong values.

```suggestion
    parser.add_argument("-dn", "--detector_model_name", help="Name of the trained model",
                        type=str, default="tanet_car_xyres_16", choices =["tanet_car_xyres_16"])
```